nvbench/CMakeLists.txt

```diff
@@ -64,6 +69,15 @@ set_target_properties(nvbench PROPERTIES
   LIBRARY_OUTPUT_DIRECTORY "${NVBench_LIBRARY_OUTPUT_DIR}"
   RUNTIME_OUTPUT_DIRECTORY "${NVBench_EXECUTABLE_OUTPUT_DIR}"
 )
+if (NVBench_ENABLE_CUPTI)
+  if (WIN32)
+    message("Don't forget to add ${CUDAToolkit_LIBRARY_ROOT}/extras/CUPTI/lib64 in your PATH!")
```

Comment:
@robertmaynard Do you know anything about this?

@senior-zero mentioned that he had to modify `PATH` like this when linking to CUPTI on windows. Is there a way to fix this without bugging the user?

Reply:
Link to the `CUDA::cuda_driver` and `CUDA::cupti` targets instead of having the explicit paths in target_include_directories, and target_link_directories plus the usage of raw libraries ( cuda, cupti ) in target_link_libraries

Reply:
#46 adds NVML support and fixes a lot of the CMake issues in this PR:

- Fixes/WARs the dll PATH issues without requiring users to modify PATH
- Adds an auto-generated `nvbench/config.cuh` header instead of relying on compiler definitions.

It should be fairly easy to extend the approaches used in that PR here. I'd suggest rebasing this on #46 so we can reuse the cmake machinery.